crates/miden-lib/asm/account_components/rpo_falcon_512_procedure_acl.masm

```diff
@@ -16,61 +16,115 @@ const.NUM_AUTH_TRIGGER_PROCS_SLOT=1
 # The slot where the map of auth trigger procedure roots is stored.
 const.AUTH_TRIGGER_PROCS_MAP_SLOT=2
 
-#! Authenticate a transaction using the Falcon signature scheme only if specified auth trigger
-#! procedures were called.
+#! Authenticate a transaction using the Falcon signature scheme based on procedure calls and note usage.
 #!
-#! This authentication procedure checks if any of the trigger procedures were called during the
-#! transaction. If none were called, authentication is skipped. If at least one was called,
-#! the standard RpoFalcon512 signature verification is performed.
-#! Nonce is incremented in both cases.
+#! This authentication procedure checks:
+#! 1. If any of the trigger procedures were called during the transaction
+#! 2. If input notes were consumed and allow_unauthorized_input_notes is false
+#! 3. If output notes were created and allow_unauthorized_output_notes is false
 #!
-#! Inputs:  [pad(16)]
+#! If any of these conditions are true, standard RpoFalcon512 signature verification is performed.
+#! Otherwise, only the nonce is incremented.
+#!
+#! Inputs:  [AUTH_ARGS, pad(16)]
```

Comment:
nit: technically this should be `[AUTH_ARGS, pad(12)]` because this procedure is expected to be invoked using the `call` instruction which cannot access more than 16 elements of the stack.

Not that `dropw` will still result in `[pad(16)]` because zeros will be shifted in from the right.